bots/images/scripts/openshift.setup

```diff
@@ -11,6 +11,8 @@ echo foobar | passwd --stdin root
 nmcli con add con-name "static-eth1" ifname eth1 type ethernet ip4 "10.111.112.101/20" gw4 10.111.112.1 ipv4.dns "10.111.112.1"
 nmcli con up "static-eth1"
 
+echo "10.111.112.101     f1.cockpit.lan" >> /etc/hosts
```

Comment:
This PR does not rebuild the openshift image. Was that commit meant to be in this PR as the *future* openshift rebuild will require it, or is it a local debugging leftover? Would be nice to add some rationale here. (Fine to add the commit message during merging)

Reply:
With out this the tests don't pass on a 3.7 origin build, because 3.7 changes what IP address get selected by the API server.

I did not push a rebuilt openshift image because 3.7 is still in RC and we generally keep to released versions.

Reply:
The other patch needed to actually build the openshift image is:

```
diff --git a/bots/images/scripts/openshift.setup b/bots/images/scripts/openshift.setup
index 69f005052..bce3a1474 100755
--- a/bots/images/scripts/openshift.setup
+++ b/bots/images/scripts/openshift.setup
@@ -241,7 +241,8 @@ docker push registry:5000/pizzazz/monster
 oc config use-context default/10-111-112-101:8443/system:admin
 
 # Some roles for testing against
-oc patch --namespace=marmalade policybinding ':default' -p '{"roleBindings":[{"name":"admin","roleBinding":{"metadata":{"name":"admin","namespace":"marmalade"},"userNames":["scruffy"],"groupNames":null,"subjects":[{"kind":"User","name":"scruffy"}],"roleRef":{"name":"admin"}}},{"name":"registry-editor","roleBinding":{"metadata":{"name":"registry-editor","namespace":"marmalade","resourceVersion":"1"},"userNames":["scruffy","amanda"],"groupNames":null,"subjects":[{"kind":"User","name":"scruffy"},{"kind":"User","name":"amanda"}],"roleRef":{"name":"registry-editor"}}},{"name":"registry-viewer","roleBinding":{"metadata":{"name":"registry-viewer","namespace":"marmalade","resourceVersion":"1"},"userNames":["scruffy","tom","amanda"],"groupNames":["sports"],"subjects":[{"kind":"User","name":"scruffy"},{"kind":"User","name":"tom"},{"kind":"User","name":"amanda"},{"kind":"Group","name":"sports"}],"roleRef":{"name":"registry-viewer"}}}]}'
+printf '{"kind":"List","apiVersion":"v1","items":[{"kind":"RoleBinding","metadata":{"name":"registry-editor","namespace":"marmalade","resourceVersion":"1"},"userNames":["scruffy","amanda"],"groupNames":null,"subjects":[{"kind":"User","name":"scruffy"},{"kind":"User","name":"amanda"}],"roleRef":{"name":"registry-editor"}},{"kind":"RoleBinding","metadata":{"name":"registry-viewer","namespace":"marmalade","resourceVersion":"1"},"userNames":["scruffy","tom","amanda"],"groupNames":["sports"],"subjects":[{"kind":"User","name":"scruffy"},{"kind":"User","name":"tom"},{"kind":"User","name":"amanda"},{"kind":"Group","name":"sports"}],"roleRef":{"name":"registry-viewer"}}]}' | oc create -f -
+oc patch rolebinding/admin --namespace=marmalade -p '{"kind": "RoleBinding", "metadata":{"name":"admin","namespace":"marmalade"},"userNames":["scruffy"],"groupNames":null,"subjects":[{"kind":"User","name":"scruffys"}],"roleRef":{"name":"admin"}}'
 
 # For testing the Cockpit OAuth client
 printf '{"kind":"OAuthClient","apiVersion":"v1","metadata":{"name":"cockpit-oauth-devel"},"respondWithChallenges":false,"secret":"secret","allowAnyScope":true,"redirectURIs":["http://localhost:9001"] }' | oc create -f -
```

But that needs to wait till we drop 3.6.

Reply:
FTR: test/verify/kubelib.py even does that explicitly:

    m.execute("echo '10.111.112.101  f1.cockpit.lan' >> /etc/hosts")

Right now, `f1.cockpit.lan` resolves to 172.27.0.15 on the current openshift image, which isn't really less broken than another non-localhost IP. So I'm convinced that this is backwards compatible.